### PR TITLE
Loriab optking

### PR DESCRIPTION
**Pull request by AlexHeide**
## Description

Update to https://github.com/MolSSI/QCEngine/pull/166. Adds a simple procedure for running optking. Tested locally with https://github.com/psi-rking/optking/pull/38.

## Changelog description

Add optking procedure

## Status

- [x] Code base linted
- [ ] Ready to go

### CODE REVIEW

File: qcengine/procedures/geometric.py
Comment (dgasmith) [inline qcengine/procedures/geometric.py]: Unsure why this file was changed? Please consider reverting.

File: qcengine/tests/test_procedures.py
Comment (dgasmith) [inline qcengine/tests/test_procedures.py]: No need for marks to be a list here. Can we write out `optimizer`?

File: qcengine/procedures/geometric.py
Comment (AlexHeide) [reply qcengine/procedures/geometric.py]: I believe that was part of #166. I assumed the purpose was to unify the two procedures. Happy to revert if you'd like

File: qcengine/tests/test_procedures.py
Comment (AlexHeide) [reply qcengine/tests/test_procedures.py]: Fixed

**Comment by AlexHeide** [comment]
Optking is not on either conda forge or PyPI. If one of those is recommended I would be happy to explore adding optking to either. We should be stable moving forward.

File: qcengine/procedures/geometric.py
Comment (loriab) [reply qcengine/procedures/geometric.py]: Yeah, pretty sure that's part of my campaign to help readability by having the the var names for the object be `_model` and the dict be `_data`.

Comment (dgasmith) [reply qcengine/procedures/geometric.py]: Ok, good to leave it.

**Comment by dgasmith** [comment]
Be best to release on PyPI + Conda before merging so we can add this to our CI. Poke us on #help in QCArchive Slack.

File: qcengine/procedures/optking.py
Comment (loriab) [inline qcengine/procedures/optking.py]: `-c psi4/label/dev` for the immediate future, not that I can get it uploaded to either.

File: qcengine/procedures/optking.py
Comment (loriab) [inline qcengine/procedures/optking.py]: JobConfig changed to TaskConfig sometime

**Comment by loriab** [comment]
Also, lint (ping me with any problems). And add optking to https://github.com/MolSSI/QCEngine/blob/master/devtools/conda-envs/psi-nightly.yaml

File: qcengine/procedures/optking.py
Comment (loriab) [inline qcengine/procedures/optking.py:31]: can use `self.found()` here like the programs do to avoid repetition.

File: qcengine/procedures/optking.py
Comment (loriab) [inline qcengine/procedures/optking.py:23]: please go ahead and add a `def version` here like the programs do

**Comment by AlexHeide** [comment]
I think this is ready to go. Can I get another review on this PR?

**Review by loriab — APPROVED** [review]
This looks in excellent shape to me. Thank you! And I apologize for the delay.

I make a new conda package from master, `_4`, and reran the test with latest psi. It would be good to get an optking version minted for packaging.

**Pull request merged**
